### PR TITLE
Add schema support

### LINKED ISSUE #23: Add support for schemas
Currently, there is no support for the schema API (https://www.elastic.co/guide/en/app-search/current/schema.html)

Is this something that would be accepted if a PR was made for adding support?

### PR DESCRIPTION
**Pull request by ckeboss**
Fixes #23

### CODE REVIEW

**Comment by ckeboss** [comment]
Signed CLA

**Comment by ckeboss** [comment]
Any update?

**Comment by JasonStoltz** [comment]
Hey @ckeboss, I have a long backlog of things to get through so I'm trying my best to get to this. Is this blocking you?

In the meantime, do you mind updating your branch with `master`? 

Thanks so much for the contribution and sorry for the delay.

File: fixtures/localhost-3002/update-schema [hunk 1]
```diff
@@ -0,0 +1,24 @@
+POST /api/as/v1/engines/swiftype-api-example/schema
+x-swiftype-client: elastic-app-search-node
+x-swiftype-client-version: 7.7.0
```
Comment (JasonStoltz) [inline fixtures/localhost-3002/update-schema:3]:
```suggestion
x-swiftype-client-version: 7.8.0
```

To fix specs

File: fixtures/localhost-3002/get-schema [hunk 2]
```diff
@@ -0,0 +1,24 @@
+GET /api/as/v1/engines/swiftype-api-example/schema
+x-swiftype-client: elastic-app-search-node
+x-swiftype-client-version: 7.7.0
```
Comment (JasonStoltz) [inline fixtures/localhost-3002/get-schema:3]:
```suggestion
x-swiftype-client-version: 7.8.0
```

To fix specs

File: fixtures/localhost-3002/get-schema [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+content-type: application/json; charset=utf-8
+vary: Origin
+etag: W/"ac828da6380c08d10df2df85cf73c250"
+cache-control: max-age=0, private, must-revalidate
+x-request-id: 952ae812-c1fc-4292-bff5-e17276f6ea10
+x-runtime: 0.042176
+connection: close
+server: Jetty(9.2.29.v20191105)
+
+{"location":"geolocation","created_date":"date","title":"text","url":"text","views":"number"}
```
Comment (JasonStoltz) [inline fixtures/localhost-3002/get-schema:24]: Add a new line please

File: fixtures/localhost-3002/update-schema [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+content-type: application/json; charset=utf-8
+vary: Origin
+etag: W/"5669e9ad4cde52b649ba90479fff9e1f"
+cache-control: max-age=0, private, must-revalidate
+x-request-id: f80545b5-9269-43e7-844d-5dae39c4c172
+x-runtime: 0.184414
+connection: close
+server: Jetty(9.2.29.v20191105)
+
+{"location":"geolocation","created_date":"date","title":"text","url":"text","views":"text"}
```
Comment (JasonStoltz) [inline fixtures/localhost-3002/update-schema:24]: Add a new line please

File: README.md [hunk 5]
```diff
@@ -329,6 +329,32 @@ client
...
+
+```javascript
+const engineName = 'favorite-videos'
+const schema = {
+  views: 'number',
+  created_at: 'date'
+}
+
+client
+  .updateSchema(engineNames, schema)
```
Comment (JasonStoltz) [inline README.md:353]:
```suggestion
  .updateSchema(engineName, schema)
```

**Comment by Christophvh** [comment]
I would love to use this feature, is there something that needs to be done? Like documenting this or fixing something?

**Comment by ckeboss** [comment]
I will make the requested changes today

**Comment by Christophvh** [comment]
@ckeboss can I help with something to get this merged?

**Comment by ckeboss** [comment]
@Christophvh sorry for the delay, I have now made the changes @JasonStoltz requested.

**Comment by Christophvh** [comment]
@JasonStoltz Are you able to look a this? It's pretty blocking for a project we are working on.

**Comment by JasonStoltz** [comment]
@Christophvh Sorry for the delay, we will review this shortly.

**Comment by ckeboss** [comment]
Any update to this?

**Comment by JasonStoltz** [comment]
Sorry for the delay @ckeboss. I'm closing this and merging in #34 which includes a fix for specs. Thank you for the contribution!

**Pull request closed without merging**